migrations/2025-02-07-105516_remove-aggregate-tables/up.sql

```diff
@@ -0,0 +1,256 @@
+CREATE INDEX idx_comment_aggregates_score ON comment USING btree (score DESC);
+
+-- merge post_aggregates into post table
+ALTER TABLE post
+    ADD COLUMN comments bigint NOT NULL DEFAULT 0,
+    ADD COLUMN score bigint NOT NULL DEFAULT 0,
+    ADD COLUMN upvotes bigint NOT NULL DEFAULT 0,
+    ADD COLUMN downvotes bigint NOT NULL DEFAULT 0,
+    ADD COLUMN newest_comment_time_necro timestamp with time zone NOT NULL DEFAULT now(),
+    ADD COLUMN newest_comment_time timestamp with time zone NOT NULL DEFAULT now(),
+    ADD COLUMN hot_rank double precision NOT NULL DEFAULT 0.0001,
+    ADD COLUMN hot_rank_active double precision NOT NULL DEFAULT 0.0001,
+    ADD COLUMN controversy_rank double precision NOT NULL DEFAULT 0,
+    ADD COLUMN instance_id int NOT NULL DEFAULT 0 REFERENCES instance (id) ON UPDATE CASCADE ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED,
+    ADD COLUMN scaled_rank double precision NOT NULL DEFAULT 0.0001,
+    ADD COLUMN report_count smallint NOT NULL DEFAULT 0,
+    ADD COLUMN unresolved_report_count smallint NOT NULL DEFAULT 0;
+
+UPDATE
+    post
+SET
+    comments = pa.comments,
+    score = pa.score,
+    upvotes = pa.upvotes,
+    downvotes = pa.downvotes,
+    newest_comment_time_necro = pa.newest_comment_time_necro,
+    newest_comment_time = pa.newest_comment_time,
+    hot_rank = pa.hot_rank,
+    hot_rank_active = pa.hot_rank_active,
+    controversy_rank = pa.controversy_rank,
+    instance_id = pa.instance_id,
+    scaled_rank = pa.scaled_rank,
+    report_count = pa.report_count,
+    unresolved_report_count = pa.unresolved_report_count
+FROM
+    post_aggregates AS pa
+WHERE
+    post.id = pa.post_id;
+
+DROP TABLE post_aggregates;
+
+-- Note, removed `post_id DESC` from all these
+CREATE INDEX idx_post_aggregates_community_active ON post USING btree (community_id, featured_local DESC, hot_rank_active DESC, published DESC);
+
+CREATE INDEX idx_post_aggregates_community_controversy ON post USING btree (community_id, featured_local DESC, controversy_rank DESC);
+
+CREATE INDEX idx_post_aggregates_community_hot ON post USING btree (community_id, featured_local DESC, hot_rank DESC, published DESC);
+
+CREATE INDEX idx_post_aggregates_community_most_comments ON post USING btree (community_id, featured_local DESC, comments DESC, published DESC);
+
+CREATE INDEX idx_post_aggregates_community_newest_comment_time ON post USING btree (community_id, featured_local DESC, newest_comment_time DESC);
+
+CREATE INDEX idx_post_aggregates_community_newest_comment_time_necro ON post USING btree (community_id, featured_local DESC, newest_comment_time_necro DESC);
+
+--CREATE INDEX idx_post_aggregates_community_published on post USING btree (community_id, featured_local DESC, published DESC);
+--CREATE INDEX idx_post_aggregates_community_published_asc on post USING btree (community_id, featured_local DESC, reverse_timestamp_sort (published) DESC);
```

Comment:
I don't see these ones being created before.

Reply:
Strange Im sure those were throwing errors before but now they can all be added. Maybe because of the renaming above, but anyway the indices should be dropped when the table is dropped.